src/lib/util/permLevels.js

```diff
@@ -33,26 +43,28 @@ class PermissionLevels {
 	 * @returns {PermissionLevels} This permission levels
 	 */
 	addLevel(level, brk, check) {
-		if (this.levels.has(level)) throw new Error(`Level ${level} is already defined`);
-		this.levels.set(level, { break: brk, check: check });
-		return this;
+		return this.set(level, { break: brk, check });
 	}
 
 	/**
-	 * The current valid permissions structure
-	 * @readonly
-	 * @type {validPermStructure}
+	 * Checks if all permission levels are valid
+	 * @return {boolean}
 	 */
-	get structure() {
-		const structure = [];
-		for (let i = 0; i < 11; i++) {
-			const myLevel = this.levels.get(i);
-			if (myLevel) structure.push(myLevel);
-			else structure.push({ break: false, check: () => false });
-		}
-		return structure;
+	isValid() {
+		/* eslint-disable no-multi-spaces, arrow-body-style, operator-linebreak, indent, no-mixed-spaces-and-tabs */
+		return every(this)((level, index) => {
+			// Trick to replace a switch case or if else. Condition on the left, what's executed on the right. Last one is default.
```

Comment:
isValid should always return a boolean: Please use standard ```javascript
if (condition) return false;
if (condition) return false;
return true;
```
additionally you could add a debug method, to return a string telling you what entry isn't valid.